card component now working and added to every page type.

danielfmiranda opened the pull request:
Closes #7139

**Link to sample test pages**:
**Card Components on article page, under a publication page:** https://foundation-s-7139-card--rrq92h.herokuapp.com/en/publication-page-with-child-article-pages/fixed-title-article-page/
**Card Components on initiative page:** https://foundation-s-7139-card--rrq92h.herokuapp.com/en/initiatives/summer-fill-with/


**Notes:** 

- the wagtail_ci test is failing as I have not added the migrations yet. Once I get approval, I will update them to ensure they are in order.


## Checklist

**Changes in Models:**
- [x] [Are my changes backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/master/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations).

danielfmiranda commented:
Hi @kristinashu! I had a question regarding the two card minimum limit. 

It appears that implementing this might be more involved than I had anticipated. For the time being, I have added help text to the card grid component in the CMS letting the user know that we recommend a minimum of 2 cards, like so: 

<img width="367" alt="image" src="https://user-images.githubusercontent.com/18314510/128070815-49162eca-6f75-48bf-af57-34f7ac9c8f2c.png">

I understand that this feature has a deadline, so I tagged you in this PR in case you would like to take a look at the rest of the of the component as it is pretty much done, and to also ask if the "help text" would suffice to meet the ask of the ticket, or if we have time to implement custom validation through the admin panel. 

Thank you!

kristinashu commented:
This looks great @danielfmiranda! I think the hint text is enough. The layout doesn't look too bad with only one card so no need to put in the effort to make it a requirement. 

I'm happy to move forward with it as is but do have a few questions.

1) Is there a reason for calling it "Grid items"? My initial though would be to call it "Cards" but don't have a clear reason why.

2) If it isn't too much work, could you markthe required fields as `*`?

3) Guessing this is a bigger lift so it could be done as a follow-up. The cards are too narrow when the page is narrow width (like your example here https://foundation-s-7139-card--rrq92h.herokuapp.com/en/initiatives/summer-fill-with/). Could the cards either be only 2 up at narrower widths or could the component break out of the narrow width and always be wide width (like you've done for wide width images)? Screen cap from prod of the narrow cards on a narrow width page:
![image](https://user-images.githubusercontent.com/1068783/128074100-7d53629e-e380-4e90-b59e-08c920d730e1.png)

danielfmiranda commented on network-api/networkapi/wagtailcustomization/static/css/admin.css:
```diff
@@ -53,3 +53,8 @@ fieldset ul li div label {
   vertical-align: middle;
 }
+
+/* Displaying red asterisk for required fields. */
```
I noticed that listblocks were not displaying the red asterisks for required fields in the cms, and found that in the wagtail css it was being set to "display:none" for some reason. 

Upon further research I found this PR for wagtail https://github.com/wagtail/wagtail/pull/6715, which mentions the same problem and a fix.

However, since we are not using the latest version of wagtail yet and the PR was pretty recent, I added this piece of CSS to make sure required fields display properly for the time being.

danielfmiranda commented:
Hi @kristinashu! Thank you for the review and feedback, I agree it is a good point that the change from grid_items to cards might make things easier to read. 

I have updated the name in the CMS from grid_items to cards, and have also now added the red asterisks. 
Per our conversation on slack, I will create a design ticket for item 3, so we can figure out which route we would like to take regarding the width of the cards. 

Thanks!

kristinashu reviewed (approved):
Design looks good!

danielfmiranda commented:
Hi @KalobTaulien,

Thanks for taking the time to review this PR! I have implemented your suggested changes and re-requested review. 

Regarding the external vs internal URL question, I am not 100% sure about the use case. Maybe @kristinashu would know if the stakeholders would like the option to be able to link anywhere, or if they would just like to link to other pages on the foundation site? 

Thank you!

kristinashu commented:
Good point about the url links - staff will need to be able to link to both internal and external sites/pages.

danielfmiranda commented on network-api/networkapi/wagtailpages/pagemodels/customblocks/card_grid.py:
```diff
@@ -0,0 +1,41 @@
+        icon = 'placeholder'
+        template = 'wagtailpages/blocks/card_grid_block.html'
+
+        
```
Fixed in latest rendition!

danielfmiranda commented:
@kristinashu Great! We will keep it as a CharBlock then so the users can link to any site, internal or external

danielfmiranda commented on network-api/networkapi/wagtailpages/pagemodels/customblocks/card_grid.py:
```diff
@@ -0,0 +1,41 @@
+        help_text='Body text of the card.'
+    )
+
+    link_url = blocks.CharBlock(
```
Thanks for bringing this up Kalob! I reached out to Kristina and she mentioned that the links should have the ability to be internal or external.  So I will resolve this conversation since we are already using charblock

danielfmiranda commented on network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/card_grid_block.html:
```diff
@@ -0,0 +1,25 @@
+            {% image card.image fill-350x197 as img %}
+            <div class="col-sm-12 {% if count > 2 %} col-lg-4 col-md-6 {% else %} col-md-6 {% endif %} mb-4">
+              <div class="card-regular h-100 d-flex flex-column ">
+                <img src={{ img.url }} alt={{ img.alt_text }} />
```
Fixed in latest push
